library/core/src/option.rs

```diff
@@ -1305,6 +1305,46 @@ impl<T> Option<T> {
         None
     }
 
+    /// Retains the value only if the predicate evaluates to true on a mutable reference to the
+    /// value.
+    ///
+    /// In other words, replaces `self` with `None` if the predicate returns `false`.
+    /// This method operates similar to [`Option::filter`] but in place.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(option_retain)]
+    ///
+    /// let mut x = Some(42);
+    ///
+    /// x.retain(|v| if *v == 42 {
+    ///     *v += 1;
+    ///     true
+    /// } else {
+    ///     false
+    /// });
+    /// assert_eq!(x, Some(43));
+    ///
+    /// x.retain(|v| *v == 42);
+    /// assert_eq!(x, None);
+    /// ```
+    #[inline]
+    #[unstable(feature = "option_retain", issue = "98934")]
+    #[rustc_const_unstable(feature = "const_option_retain", issue = "98934")]
+    pub const fn retain<P>(&mut self, predicate: P)
```

Comment:
It looks like the ACP was accepted as take_if, not retain? Can you update accordingly?